tests/unit/test_dovecot.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""dovecot service unit tests."""
+
+from unittest.mock import Mock, patch
+
+import pytest
+
+from reactive import dovecot, utils
+
+
+class TestConstructDovecotConfigFileContent:
+    @pytest.mark.parametrize(
+        ("enable_smtp_auth"),
+        [
+            pytest.param(True, id="smtp_auth_enabled"),
+            pytest.param(False, id="smtp_auth_disabled"),
+        ],
+    )
+    @patch("reactive.dovecot.utils.render_jinja2_template")
+    def test_content_construction(
+        self,
+        mock_render: Mock,
+        enable_smtp_auth: bool,
+    ) -> None:
+        """
+        arrange: Given different values for enabling SMTP auth.
+        act: Call construct_dovecot_config_file_content.
+        assert: The Jinja2 renderer is called with the correctly formatted context.
+        """
+        # Arrange
+        dovecot_users_path = "/etc/dovecot/users"
+        expected_context = {
+            "JUJU_HEADER": utils.JUJU_HEADER,
+            "passdb_driver": "passwd-file",
+            "passdb_args": f"scheme=CRYPT username_format=%u {dovecot_users_path}",
+            "path": "/var/spool/postfix/private/auth",
+            "smtp_auth": enable_smtp_auth,
+        }
+
+        # Act
+        result = dovecot.construct_dovecot_config_file_content(
+            dovecot_users_path, enable_smtp_auth
+        )
+
+        # Assert
+        mock_render.assert_called_once_with(expected_context, "templates/dovecot_conf.tmpl")
+        assert result == mock_render.return_value
```

Comment:
You are comparing the result with what the mock will return, and the mock is actually called inside `construct_dovecot_config_file_content`, meaning you are comparing the same string fetched from different places. You'll need to store a file with the expected result and compare `result` with the file content

Reply:
As a consequence, I don't think ` mock_render.assert_called_once_with(expected_context, "templates/dovecot_conf.tmpl")` provides any additional value

Reply:
I think you missed my point here